compiler/src/iree/compiler/Codegen/Common/TensorDynamicDimAnalysis.cpp

```diff
@@ -0,0 +1,236 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/TensorDynamicDimAnalysis.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "iree/compiler/Dialect/Util/Analysis/IntegerDivisibilityAnalysis.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Analysis/DataFlow/DeadCodeAnalysis.h"
+#include "mlir/Analysis/DataFlow/IntegerRangeAnalysis.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Interfaces/DestinationStyleOpInterface.h"
+
+#define DEBUG_TYPE "iree-codegen-dynamic-dim-analysis"
+
+namespace mlir::iree_compiler {
+
+//===---------------------------------------------------------------------===//
+// Helper function to update tensor dynamic dimension info
+//===---------------------------------------------------------------------===//
+
+static void
+updateRangeInfo(TensorDynamicDimAnalysis::TensorDimRangeInfo &rangeInfo,
+                Value v, unsigned dim, const ConstantIntRanges &range) {
+  assert(!rangeInfo.contains({v, dim}) &&
+         "overwriting existing dim range info");
+  rangeInfo.insert({{v, dim},
+                    ConstantIntRanges(range.umin(), range.umax(), range.smin(),
+                                      range.smax())});
+}
+
+static void updateDivisibilityInfo(
```

Comment:
`updateDivisibilityInfo` -> `initializeDivisibilityInfo` for the same reason.

Reply:
ditto as above. The `divisibilityInfo` is being updated.